tools/build/raku-ast-compiler.nqp

```diff
@@ -56,20 +61,25 @@ grammar RakuASTParser {
         [$<raw>=[is raw]]?
     }
 
+    token sigil {
+        '$' | '@' | '%'
+    }
+
     token nqp-code {
         # We want to do some minor transforms on the NQP code, so just sorta
         # tokenize it. If it's good enough for the C preproc... :-)
         (
         | <name>
         | <attribute>
-        | $<variable>=['$' <.identifier>]
+        | $<variable>=[<.sigil> '*'? <.identifier>]
         | <string>
         | ['/' <-[/]>+ '/' || '//' || '/' <?before \s* [\d | '$']>] # regex or // operator
         | $<numeric>=[ \d+ ['.' \d*]? [<[eE]> \d+]? ]
-        | $<paren>='(' <nqp-code> [ ')' || <.panic('Missing )')> ]
-        | $<brace>='{' <nqp-code> [ '}' || <.panic('Missing }')> ]
+        | $<paren>='(' <nqp-code> [ ')' || {} <.panic('Missing ) for opening ( at line ' ~ self.line-of($<paren>))> ]
+        | $<brace>='{' <nqp-code> [ '}' || {} <.panic('Missing } for opening { at line ' ~ self.line-of($<brace>))> ]
+        | $<brckt>='[' <nqp-code> [ ']' || {} <.panic('Missing ] for opening [ at line ' ~ self.line-of($<brckt>))> ]
```

Comment:
While it's nice that it's just five letters, I'd prefer the full name (i.e., 'bracket'). It just makes it harder to search for if anyone ever wants to find the implementation.

Reply:
You make me cry in sorrow! :) I'm not a big fan of terrible shortnames, but this time I'm addicted. :)